odoo/addons/base/security/base_security.xml

```diff
@@ -170,6 +170,28 @@
             <field eval="True" name="global"/>
             <field name="domain_force">[('company_ids', 'in', company_ids)]</field>
         </record>
+
+        <!-- rules for API token -->
+        <record id="api_key_user" model="ir.rule">
+            <field name="name">Users can read and delete their own keys</field>
+            <field name="model_id" ref="model_res_users_apikeys"/>
+            <field name="domain_force">[('user_id', '=', user.id)]</field>
+            <field name="groups" eval="[
+                (4, ref('base.group_public')),
```

Comment:
Since #28765 we can have a FALSE_LEAF domain for group_public

Reply:
(so group_public should have an ir.rule forbidding all accesses, that way even if someone or breaks ir.model.access they still won't have access to api keys)